{{cookiecutter.project_shortname}}/src/lib/components/{{cookiecutter.component_name}}.react.js

```diff
@@ -1,5 +1,9 @@
-{%- if cookiecutter.use_async == "True" -%}
-    {%- include 'cookiecutter_templates/AsyncComponent.react.js' -%}
+{%- if "Async" in cookiecutter.component_type and "Class" in cookiecutter.component_type -%}
+    {%- include 'cookiecutter_templates/AsyncClassComponent.react.js' -%}
+{%- elif "Async" in cookiecutter.component_type and "Function" in cookiecutter.component_type -%}
+    {%- include 'cookiecutter_templates/AsyncFunctionComponent.react.js' -%}
+{%- elif "Class" in cookiecutter.component_type -%}
+    {%- include 'cookiecutter_templates/ClassComponent.react.js' -%}
 {%- else -%}
-    {%- include 'cookiecutter_templates/Component.react.js' -%}
-{%- endif -%}
\ No newline at end of file
+    {%- include 'cookiecutter_templates/FunctionComponent.react.js' -%}
+{%- endif -%}
```

Comment:
The `FunctionComponent` or `ClassComponent` should also go in the fragment.
Here it should be:
```
if use_async:
    AsyncComponent
elif component_type == "FunctionComponent":
    FunctionComponent
else:
   ClassComponent
```

Reply:
I might be looking in the wrong place, but there seems to be a [standing PR](https://github.com/cookiecutter/cookiecutter/pull/1645) to enable conditional prompt questions based on the answer of another so I wasn't able to show the `component_type` question based on whether `use_async` is set to false or not. Let me know if there is another way to do it, otherwise both questions will show up but the template creation will depend on the user choice as you suggested.